examples/docker-client/client-code/client.py

```diff
@@ -0,0 +1,114 @@
+import warnings
+from collections import OrderedDict
+
+import flwr as fl
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+from torchvision.datasets import CIFAR10
+from torchvision.transforms import Compose, Normalize, ToTensor
+from tqdm import tqdm
+
+
+# #############################################################################
+# 1. Regular PyTorch pipeline: nn.Module, train, test, and DataLoader
+# #############################################################################
+
+warnings.filterwarnings("ignore", category=UserWarning)
+DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+
+class Net(nn.Module):
+    """Model (simple CNN adapted from 'PyTorch: A 60 Minute Blitz')"""
+
+    def __init__(self) -> None:
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(3, 6, 5)
+        self.pool = nn.MaxPool2d(2, 2)
+        self.conv2 = nn.Conv2d(6, 16, 5)
+        self.fc1 = nn.Linear(16 * 5 * 5, 120)
+        self.fc2 = nn.Linear(120, 84)
+        self.fc3 = nn.Linear(84, 10)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.pool(F.relu(self.conv1(x)))
+        x = self.pool(F.relu(self.conv2(x)))
+        x = x.view(-1, 16 * 5 * 5)
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        return self.fc3(x)
+
+
+def train(net, trainloader, epochs):
+    """Train the model on the training set."""
+    criterion = torch.nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(net.parameters(), lr=0.001, momentum=0.9)
+    for _ in range(epochs):
+        for images, labels in tqdm(trainloader):
+            optimizer.zero_grad()
+            criterion(net(images.to(DEVICE)), labels.to(DEVICE)).backward()
+            optimizer.step()
+
+
+def test(net, testloader):
+    """Validate the model on the test set."""
+    criterion = torch.nn.CrossEntropyLoss()
+    correct, loss = 0, 0.0
+    with torch.no_grad():
+        for images, labels in tqdm(testloader):
+            outputs = net(images.to(DEVICE))
+            labels = labels.to(DEVICE)
+            loss += criterion(outputs, labels).item()
+            correct += (torch.max(outputs.data, 1)[1] == labels).sum().item()
+    accuracy = correct / len(testloader.dataset)
+    return loss, accuracy
+
+
+def load_data():
+    """Load CIFAR-10 (training and test set)."""
+    trf = Compose([ToTensor(), Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
+    trainset = CIFAR10("./data", train=True, download=True, transform=trf)
+    testset = CIFAR10("./data", train=False, download=True, transform=trf)
+    return DataLoader(trainset, batch_size=32, shuffle=True), DataLoader(testset)
```

Comment:
Good point! I will add that.

Reply:
I tried to run the [quickstart-pytorch/client.py](https://github.com/adap/flower/blob/b43cf12d9e1438ca4bff465346a36b66c76f250b/examples/quickstart-pytorch/client.py#L72) example, but I encountered some problems.

We create two `ArgumentParser` instances (one in the client example, the other one in the `run_client` function in `src/py/flwr/client/app.py`), and both conflict with each other because they receive unrecognized arguments. For `run_client` it is the `--node-id` argument and `--insecure`, `--server` for the `ArgumentParser` in the client example. We can fix this by replacing `parse_args` with [parse_known_args](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.parse_known_args). 

Another problem is that I get a `'NoneType' object has no attribute 'partition'` error in the client when the driver requests the initial parameters. Does this client example work with the new driver/fleet server?

Reply:
oh, right. it won't work yet if you use the callable logic and `flower-client`.  We can either hardcode the client id so we extract the partition, or leave the dataset logic as is and revisit the example in the future. Alternatively, we don't use `flower-client` and therefore do something like `python client.py` that internally calls `flwr.client.start_client`. This approach is just like in `quickstart-pytorch/client.py`. 

@danieljanes, do you have a preference? new approach which is nicer but a bit less versatile atm, or standard approach and more flexible?